pandas/core/series.py

```diff
@@ -5642,18 +5642,15 @@ def to_timestamp(
         -------
         Series with DatetimeIndex
         """
-        new_values = self._values
-        if copy:
-            new_values = new_values.copy()
+        new_obj = self.copy(deep=copy)
 
         if not isinstance(self.index, PeriodIndex):
             raise TypeError(f"unsupported Type {type(self.index).__name__}")
         new_index = self.index.to_timestamp(freq=freq, how=how)
-        return self._constructor(new_values, index=new_index).__finalize__(
-            self, method="to_timestamp"
-        )
+        setattr(new_obj, "index", new_index)
+        return new_obj
```

Comment:
I am not fully sure, but we _might_ want to keep the `__finalize__` call? (although of course not a new object was constructed here, that already happened now in the copy() call)

Reply:
Ah, I see this is how it is done for the `DataFrame` implementation, so that's probably OK then